libraries/chain/controller.cpp

```diff
@@ -549,7 +549,7 @@ struct assembled_block {
                                    },
                                    [&](assembled_block_if& ab) {
                                       auto bsp = std::make_shared<block_state>(ab.bhs, std::move(ab.trx_metas),
-                                                                               std::move(ab.trx_receipts));
+                                                                               std::move(ab.trx_receipts), std::move(ab.qc));
```

Comment:
This should not be a `std::move`

Reply:
> This should not be a `std::move`

Why not? I think the constructor should take a rvalue and it should be a std::move.

Reply:
If we cache it in `block_state` then makes sense to move it, otherwise faster to just pass by value.

Reply:
We don't store it in `bloclk_state`. Do you want me to change to pass by value?

Reply:
You can leave it as is, thanks Lin. We can change it if the QC needs to be stored in block_state.